### PR TITLE
fix(mobile): fixes on language change

### PR DESCRIPTION
**Pull request by johnstef99**
This pull request fixes several problems around the flow of changing language:

1. When you change language some widgets in the pages `[SettingsSubPage, SettingsPage, LibraryPage]` stay in the old language.
2. Because across the app, you don't pass the `context.locale` to `DateFormat`, by default `Intl` uses the system's locale. This means that even after selecting a language in the immich settings, the dates will stay in the system's language.
3. I created a new provider `localeProvider` that can be used inside other providers to invalidate them whenever the locale is changed. This is needed because immich has providers that cache UI elements. I added the `localeProvider` only to the asset providers of the photos page but I am sure someone with more knowledge of the codebase can find other instances where it's needed.

Before fix:

https://github.com/user-attachments/assets/c9c7daf6-c4ae-44c5-8a1e-f2f873ad4332

After fix:
As you can see when changing language both settings and photo grid get updated.

https://github.com/user-attachments/assets/ac7f01b3-2ddf-48dc-a0c2-feb99401e897

### CODE REVIEW

**Comment by NicholasFlamy** [comment]
@johnstef99 Is PR #14035 related to one of the issues you're fixing?

**Comment by johnstef99** [comment]
Is kind of related, because the user describes this:

> Changing my phone to English does resolve the issue.

That means that `DateFormat` does not use the `context.locale` and instead uses the phone's locale. That problem is going to get fixed by this change [here](https://github.com/immich-app/immich/pull/14089/files#diff-b1af8ebfb559b57f56e8fc09be639e8c79833a6a806dca7bcb84f4d1b51f52a8R198)

**Comment by NicholasFlamy** [comment]
> Is kind of related, because the user describes this:
> 
> > Changing my phone to English does resolve the issue.
> 
> That means that `DateFormat` does not use the `context.locale` and instead uses the phone's locale. That problem is going to get fixed by this change [here](https://github.com/immich-app/immich/pull/14089/files#diff-b1af8ebfb559b57f56e8fc09be639e8c79833a6a806dca7bcb84f4d1b51f52a8R198)

Okay, just wanna be sure your changes and that PR don't conflict.

**Comment by johnstef99** [comment]
No conflicts for sure. The #14035 initialises the date formatter, but this PR make the date formatter use the selected locale from immich settings.

File: mobile/lib/pages/common/settings.page.dart [hunk 1]
```diff
@@ -46,6 +46,7 @@ class SettingsPage extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    context.locale;
```
Comment (alextran1502) [inline mobile/lib/pages/common/settings.page.dart:49]: Is this used to trigger the widget rebuild?

**Review by alextran1502 — APPROVED** [review]

**Pull request merged**